wordpress-mobile/WordPress-Android
# [Blogging prompts v1] Update editor analytics events

Fixes #16521 

To test:

- Open `Debug settings` and enable `blogging_prompts_notification_remote_field`

1 - To test `my_site_create_sheet_answer_prompt_tapped`:
- With a selected site, open "My Site" tab;
- Tap on the FAB;
- Tap on the "Answer prompt" button. The event `my_site_create_sheet_answer_prompt_tapped` should be tracked.

2 - To test updated `editor_session_start` event with `entry_point` : `bloggging_prompts_introduction`:
- Tap on the "Learn more" button in the feature introduction notification;
- Tap on the "Try it now" button to open the editor screen;
- `editor_session_start` event should be tracked with the extra parameter `entry_point` with value `bloggging_prompts_introduction`

3 - To test updated `editor_session_start` event with `entry_point`: `blogging_reminders_notification_answer_prompt `:
- On `My Site`, tap on `Menu`
- Tap on `Site Settings`
- Tap on `Reminders and prompts` and select a date and time for the reminder
- Change the device date to something after the selected date/time for the reminder
- A notification to answer the prompt should appear
- Tap on the notification and the editor should be opened
- `editor_session_start` event should be tracked with the extra parameter `entry_point` with value `blogging_reminders_notification_answer_prompt`

4 - To test updated `editor_session_start` event with `entry_point`: `my_site_card_answer_prompt `:
- With a site selected, tap on "HOME" tab
- Scroll to prompts card
- Tap on "Answer prompt" button
- `editor_session_start` event should be tracked with the extra parameter `entry_point` with value `my_site_card_answer_prompt`
 
5 - To test updated `editor_session_start` event with `entry_point`: `add_new_sheet_answer_prompt`
- With a selected site, open "My Site" tab;
- Tap on the FAB;
- Tap on the "Answer prompt" button.
- `editor_session_start` event should be tracked with the extra parameter `entry_point` with value `add_new_sheet_answer_prompt`

6 - To test updated `editor_post_published` event with `prompt_id` from blogging prompts introduction
- Tap on the "Learn more" button in the feature introduction notification;
- Tap on the "Try it now" button to open the editor screen;
- Publish the post: `editor_post_published` should be tracked with the extra parameter `prompt_id`

7 - To test updated `editor_post_published` event with `prompt_id` from blogging reminders notification
- On `My Site`, tap on `Menu`
- Tap on `Site Settings`
- Tap on `Reminders and prompts` and select a date and time for the reminder
- Change the device date to something after the selected date/time for the reminder
- A notification to answer the prompt should appear
- Tap on the notification and the editor should be opened
- Publish the post: `editor_post_published` should be tracked with the extra parameter `prompt_id`

8 - To test updated `editor_post_published` event with `prompt_id` from "My Site" card:
- With a selected site, open "My Site" tab;
- Tap on the FAB;
- Tap on the "Answer prompt" button.
- Publish the post: `editor_post_published` should be tracked with the extra parameter `prompt_id`

9 - To test updated `editor_post_published` event with `prompt_id` from "Add new" bottom sheet:
- With a selected site, open "My Site" tab;
- Tap on the FAB;
- Tap on the "Answer prompt" button.
- Publish the post: `editor_post_published` should be tracked with the extra parameter `prompt_id`

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
